Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -19,34 +19,21 @@ def file_linked_libraries(file, string)
   undef relocate_dynamic_linkage
 
   def relocate_dynamic_linkage(relocation)
-    old_prefix, new_prefix = relocation.replacement_pair_for(:prefix)
-    old_cellar, new_cellar = relocation.replacement_pair_for(:cellar)
-
     mach_o_files.each do |file|
       file.ensure_writable do
         if file.dylib?
-          id = dylib_id_for(file).sub(old_prefix, new_prefix)
-          change_dylib_id(id, file)
+          id = relocated_name_for(dylib_id_for(file), relocation)
+          change_dylib_id(id, file) if id
```

Comment:
I'm a little nervous about this change since there is a change in logic here. I don't think there is a change in semantics [*], but there might be a corner case I'm not thinking of.

[*] `dylib_id_for(file)` should never start with `old_cellar` because of `#fix_dynamic_linkage`. On the other hand, I also thought it should always start with `old_prefix`. But it doesn't -- hence the `if id` check.

Reply:
> `dylib_id_for(file)` should never start with `old_cellar` because of `#fix_dynamic_linkage`

In a case where "should never" I'd suggest checking and throwing an exception when it's not the case.

Reply:
> Just curious, do you have an example or when `dylib_id_for(file)` doesn't start with `old_prefix`?

Yes. `dylib_id_for(file)` doesn't start with `old_prefix` whenever `old_prefix == @@HOMEBREW_PREFIX@@`, since `dylib_id_for(file)` is a string that starts with `HOMEBREW_PREFIX` (i.e. `/usr/local`, etc).

> In a case where "should never" I'd suggest checking and throwing an exception when it's not the case.

Let me think about the right way to do that.